firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/messages.txt

```diff
@@ -64,144 +108,223 @@ Scenario: Message: The write method correctly generates an encoded body from the
 
 Scenario: Message: The parse method correctly updates internal message type and sets the message payload fields after decoding from the input buffer
 
-  GIVEN("A Message object constructed with StateSegment Taggedunion, default message types and a max size of 252 bytes")
+  GIVEN("A Message object default-constructed with StateSegment Taggedunion, default message types and a max size of 252 bytes")
     // setup:
     // message class type member is unchanged
     // payload is uninitialised
 
     WHEN("An empty input buffer body is parsed")
       THEN("The parse method reports invalid length status")
-      THEN("The payload Taggedunion tag is set to Unknown")
+      THEN("After the parse method is called, the type member is set to 0")
+      THEN("The payload.tag field remains unchanged")
+      THEN("the payload.values field remains unchanged")
+
+    WHEN("A body with an empty payload and 1-byte header whose value is not included in MessageTypes enum")
+      THEN("The parse method reports invalid type status")
+      THEN("After the parse method is called, The value assigned to the type member is equal to the type field of the input_buffer body's header")
+      THEN("The payload.tag field remains unchanged")
       THEN("the payload.values field remains unchanged")
 
-    WHEN("A body with an encoded payload and 1 byte header of value equal to 0 is parsed")
+    WHEN("A body with an empty payload and 1-byte header of value equal to the message descriptor array size is parsed")
+    // setup:
+    // custom descriptor array of size 4
       THEN("The parse method reports invalid type status")
       THEN("After the parse method is called, The value assigned to the type member is equal to the type field of the input_buffer body's header")
-      THEN("The payload Taggedunion tag is set to Unknown")
+      THEN("The payload.tag field is set to parameters")
```

Comment:
I don't understand the relationship between payload.tag == parameters and the size of the message descriptor array. Maybe there's some information in the test code which isn't described in the test description? (similar for the next WHEN with payload.tag == parameters_request)

Reply:
Right the information in the test code would be the header of the input_buffer which in this case is `0x04` (which is equal to the descriptor array size), the reason I created a different descriptor array rather than using the one from the backend for these two cases was that to return `invalid_type` on an array of 8 elements, `MessageTypes::includes(type)` would return `invalid_type` instead.  therefore created an array with 4 elements, and added the test cases for equal to and greater than the array size.

And also that's why the `payload.tag is set to parameters`  same is the case for the next WHEN, where the header is`0x05`. so I'm just asserting that the payload.tag parameter is updated.

Reply:
I see, so the context here is "THEN(The payload.tag field is set to parameters, _which corresponds to value 0x04_)"